.github/workflows/asan.yaml

```
@@ -1,23 +1,34 @@
 name: asan
-on: 
-  push:
+on:
   pull_request:
+  push:
+    branches: [ main ]
   schedule:
     - cron: '7 0 * * *'
+defaults:
+  run:
+    shell: bash
 
 jobs:
   asan:
     name: asan
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-jammy-latest
     steps:
       - name: deps
-        uses: ros-tooling/setup-ros@v0.2
+        uses: ros-tooling/setup-ros@v0.3
```

A new version has been released related to some problems in RMF https://github.com/ros-tooling/setup-ros/pull/507#pullrequestreview-1099653924
```suggestion
        uses: ros-tooling/setup-ros@v0.4
```